app/src/main/java/protect/card_locker/DBHelper.java

```diff
@@ -606,13 +636,20 @@ public static Cursor getLoyaltyCardCursor(SQLiteDatabase database, String filter
             }
         }
 
+        StringBuilder archiveString = new StringBuilder();
+        if(archiveFilter.equals(LoyaltyCardArchiveFilter.Unarchived))
+            archiveString.append(" WHERE " + LoyaltyCardDbIds.TABLE + "." + LoyaltyCardDbIds.ARCHIVE_STATUS + " = 0 ");
+        else{
+            archiveString.append(" ");
+        }
```

Comment:
I mean i could do that but i think that it complicates the issue, cause i would have to add another checkbox or something so that only the archived cards would be shown. I don't mind adding that but i don't really think that it would be practical, right now the user can see all the cards and distinguish which one is archived and which one is not just by seeing the icon of archived in the card, i think this way i just the easiest way cause the user would be able to "restore" (unarchive) the card that he wants. Tell me what do you prefer or if you have a better idea, i'm all ears.

Reply:
I meant more to use clearer naming of the LoyaltyCardArchiveFilter enum. Don't need an extra checkbox or anything.

How about:
LoyaltyCardArchiveFilter.All for all cards
LoyaltyCardArchiveFilter.Active for all cards not archived

Reply:
Ohh i see yeah that seems a lot easier to understand gonna change it right now

Reply:
Looks better, but there's no need to use a StringBuilder if it's just a plain string.

I'd say just move it into the rawQuery like filter is:
```
(filter.trim().isEmpty() ? " " : " AND " + LoyaltyCardDbFTS.TABLE + " MATCH ? ") +
```